src/middleware/company.authentication.middleware.ts

```diff
@@ -0,0 +1,32 @@
+import { Request, Response, NextFunction } from 'express';
+import { logger } from '../utils/logger';
+import { authMiddleware, AuthOptions } from "@companieshouse/web-security-node";
+import { CHS_URL, OVERSEAS_ENTITY_PRESENTER_URL } from '../config';
+import { getApplicationData } from "../utils/application.data";
+import { ApplicationData } from "../model";
+import { EntityNumberKey } from "../model/data.types.model";
+
+export const companyAuthentication = (req: Request, res: Response, next: NextFunction) => {
+  try {
+    logger.debugRequest(req, `COMPANY AUTHENTICATION REQUEST`);
+
+    const appData: ApplicationData = getApplicationData(req.session);
+    const entityNumber: string = appData?.[EntityNumberKey] as string;
+
+    if (entityNumber) {
+      const authMiddlewareConfig: AuthOptions = {
+        chsWebUrl: CHS_URL,
+        returnUrl: OVERSEAS_ENTITY_PRESENTER_URL,
+        companyNumber: entityNumber
+      };
+      logger.infoRequest(req, `Invoking company authentication with (${ entityNumber }) present in session`);
+      return authMiddleware(authMiddlewareConfig)(req, res, next);
+    } else {
+      logger.errorRequest(req, "No entity number to authenticate against");
```

Comment:
Should this redirect to OE number query page?

Reply:
can do that change, think it can be only tested in unit test and not locally since it will be hard to replicate the scenario

Reply:
Suggestion only. It really should not happen so maybe it's better to error.